synapse/storage/databases/main/room.py

```diff
@@ -1232,28 +1232,30 @@ async def get_partial_state_room_resync_info(
         """
         room_servers: Dict[str, PartialStateResyncInfo] = {}
 
-        rows = await self.db_pool.simple_select_list(
-            table="partial_state_rooms",
-            keyvalues={},
-            retcols=("room_id", "joined_via"),
-            desc="get_server_which_served_partial_join",
+        rows = cast(
+            List[Tuple[str, str]],
```

Comment:
`joined_via` is nullable?

Reply:
I think I added that column after the fact.

ISTR that recent postgresses have made it so that adding a new column with a non-null default is much faster than it used to be. Maybe we don't need to add nullable columns when extending a table these days?

Reply:
This is as far as I reviewed up to

Reply:
> I think I added that column after the fact.
> 
> ISTR that recent postgresses have made it so that adding a new column with a non-null default is much faster than it used to be. Maybe we don't need to add nullable columns when extending a table these days?

[PG 11's changelog](https://www.postgresql.org/docs/release/11.0/) says

> Many other useful performance improvements, including the ability to avoid a table rewrite for ALTER TABLE ADD COLUMN with a non-null column default

Reply:
I think this table is usually small so we can also probably just change it on the fly? I've filed an issue for now: #16547